src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,64 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int PASSWORD_LENGTH = 6;
+    private static final int LOGIN_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        isUserNull(user);
+        isLoginValid(user);
+        isPasswordValid(user);
+        isAgeValid(user);
+        return storageDao.add(user);
     }
```

Comment:
It's just recommendation to consider - maybe for better readability it's better to move all the methods with validation logic to one method (for example validateUser(user)) and place all these method's call in it
```suggestion
        validateUser(user);
        return storageDao.add(user);
    }
```

Reply:
I considered such a decision and I think let the mentors tell me what will be better, and I will gladly correct it if necessary